functions/src/notifications.ts

```diff
@@ -88,4 +90,56 @@ export function createActivityNotification(token: string, activity_id: string):
         },
         data: { activity_id },
     }
+}
+
+export async function newMessageHandler(snap: FirebaseFirestore.DocumentSnapshot, context: EventContext) {
+    const activity_id = context.params.activity_id
+    // @ts-ignore because it doesn't like the mock
+    const activityDoc = await Refs(database).activity(activity_id).get()
+    const activity = activityDoc.data() as Activity
+    const chatMessage = snap.data() as ChatMessage
+
+    // @ts-ignore because it doesn't like the mock
+    const tokens = await getTokensForChatNotification(activity.members, chatMessage.sender, database)
+
+    // Send a notification to each token based on the message:
+    return Promise.all(tokens.map(async (token: string) => {
+        const message = createChatNotification(token, activity_id, `New message in ${activity.title}:`, chatMessage.message)
```

Comment:
I think we should just send `activity.title` as the notification title. That's really what's important, and that way for a long activity title we can show more text in the notification.

Reply:
That sounds good :)